src/single_random_draw.rs

```diff
@@ -0,0 +1,90 @@
+#[cfg(any(test, feature = "rand"))]
+use crate::Utxo;
+
+#[cfg(any(test, feature = "rand"))]
+use rand::{seq::SliceRandom, thread_rng};
+
+/// Randomly select coins for the given target by shuffling the utxo pool and
+/// taking UTXOs until the given target is reached, or returns None if the target
+/// cannot be reached with the given utxo pool.
+/// Requires compilation with the "rand" feature.
+#[cfg(any(test, feature = "rand"))]
+#[cfg_attr(docsrs, doc(cfg(feature = "rand")))]
+pub fn select_coins_srd<T: Utxo>(target: u64, utxo_pool: &mut [T]) -> Option<Vec<T>> {
```

Comment:
What do you think about taking the RNG as a parameter, similar to [this](https://github.com/rust-bitcoin/rust-secp256k1/blob/master/src/key.rs#L201)?

Reply:
I'm not sure I grok the linked Rust, but an advantage of having the seed for randomness as an input is that it allows you to pass a static seed in tests to make them repeatable.

Reply:
I think it's a good idea if as @Xekyo mentioned that it's a way to make the tests deterministic.  Is there any other reason you ask?  In either case I'm happy to look at adding RNG as a parameter.

The `<R: rand::Rng + ?Sized>` part is probably unfamiliar when compared to other languages and hard to grok at first - however it's what Rust calls a trait bound - which simply tells the compiler that the type `R` must have the following traits (must have a known size at compile time and implement the method signatures for the abstract type rand::Rng).

Reply:
> I think it's a good idea if as @Xekyo mentioned that it's a way to make the tests to be deterministic.

Yes you can use [`StepRng`](https://docs.rs/rand/latest/rand/rngs/mock/struct.StepRng.html) for that :)!

> Is there any other reason you ask?

`thread_rng()` requires std so if someone doesn't want to use std it's nice to be able to give something else. You could also have two functions, one that takes the RNG and one that doesn't but requires an extra feature (e.g. `rand-std`). That's what I had for for [the Schnorr API of secp256k1](https://github.com/rust-bitcoin/rust-secp256k1/blob/master/src/schnorr.rs#L125). Might be an overkill for now though so I'd personally go with just taking the RNG as parameter.

Reply:
Added RNG here: https://github.com/p2pderivatives/rust-bitcoin-coin-selection/pull/22/commits/9ea6fab1c65259fc98667d7c0400a71142158275